src/main/java/org/apache/commons/io/input/QueueInputStream.java

```diff
@@ -224,4 +226,48 @@ public int read() {
         }
     }
 
+    /**
+     * Reads up to {@code length} bytes of data from the input stream into
+     * an array of bytes.  The first byte is read while honoring the timeout; the rest are read while <i>not</i> honoring
+     * the timeout. The number of bytes actually read is returned as an integer.
+     *
+     * @param b     the buffer into which the data is read.
+     * @param offset   the start offset in array {@code b} at which the data is written.
+     * @param length   the maximum number of bytes to read.
+     * @return     the total number of bytes read into the buffer, or {@code -1} if there is no more data because the
+     *              end of the stream has been reached.
+     * @throws NullPointerException If {@code b} is {@code null}.
+     * @throws IllegalStateException if thread is interrupted while waiting for the first byte.
+     * @throws IndexOutOfBoundsException if {@code offset} is negative, {@code length} is negative, or {@code length} is
+     *             greater than {@code b.length - offset}.
+     * @since 2.20.0
+     */
+    @Override
+    public int read(final byte[] b, final int offset, final int length) {
+        if (b == null) {
+            throw new NullPointerException();
+        } else if (offset < 0 || length < 0 || length > b.length - offset) {
+            throw new IndexOutOfBoundsException();
```

Comment:
_Nitpick_: Let us use the same message as the overridden method. See also https://github.com/apache/commons-io/pull/748#discussion_r2094044658

```suggestion
            throw new IndexOutOfBoundsException(
                    String.format("Range [%d, %<d + %d) out of bounds for length %d", offset, length, b.length));
```

Reply:
@ppkarwasz 
But that's _not_ what's in the superclass in Java 8. On Java 11, if you drill down, you end up in `jdk.internal.util.Preconditions.outOfBoundsMessage(String, List<Integer>)` which has the kind of message you show. I imagine subsequent Java LTS versions might yet do something else. 
Which one should we pick? If Java 11 gives more detail they must have decided that there are no security implications, so I'd be OK with that version.

Reply:
Good point!